packages/actor-http-native/lib/ActorHttpNative.ts

```diff
@@ -63,41 +63,56 @@ export class ActorHttpNative extends ActorHttp {
       const req = this.requester.createRequest(options);
       req.on('error', reject);
       req.on('response', httpResponse => {
-        httpResponse.on('error', (error: Error) => {
-          httpResponse = null;
-          reject(error);
-        });
-        // Avoid memory leak on HEAD requests
-        if (options.method === 'HEAD') {
-          httpResponse.destroy();
-        }
-        // Using setImmediate so error can be caught should it be thrown
-        setImmediate(() => {
-          if (httpResponse) {
-            // Expose fetch cancel promise
-            httpResponse.cancel = () => {
-              httpResponse.destroy();
-              return Promise.resolve(undefined);
-            };
-            // Missing several of the required fetch fields
-            const headers = new Headers(httpResponse.headers);
-            const result = <IActorHttpOutput> {
-              body: httpResponse,
-              headers,
-              ok: httpResponse.statusCode < 300,
-              redirected: options.url !== httpResponse.responseUrl,
-              status: httpResponse.statusCode,
-              // When the content came from another resource because of conneg, let Content-Location deliver the url
-              url: headers.has('content-location') ? headers.get('content-location') : httpResponse.responseUrl,
-            };
-            resolve(result);
-          }
-        });
+        processHttpResponse(httpResponse, options, resolve, reject);
       });
     });
   }
 }
 
+export function processHttpResponse(httpResponse: any, options: any, resolve: any, reject: any): void {
+  httpResponse.on('error', (error: Error) => {
+    httpResponse = null;
+    reject(error);
+  });
+  // Avoid memory leak on HEAD requests
+  if (options.method === 'HEAD') {
+    httpResponse.destroy();
+  }
+  // Using setImmediate so error can be caught should it be thrown
+  setImmediate(() => {
+    if (httpResponse) {
+      // Expose fetch cancel promise
+      httpResponse.cancel = () => {
+        httpResponse.destroy();
+        return Promise.resolve(undefined);
+      };
+      // Missing several of the required fetch fields
+
+      let headers = new Headers(httpResponse.headers);
+      if (httpResponse && httpResponse.input && httpResponse.input.headers) {
+        headers = httpResponse.input.headers;
+      }
+
+      if (httpResponse && httpResponse.rawHeaders) {
+        for (let i = 0; i + 1 < httpResponse.rawHeaders.length; i += 2) {
+          headers.set(httpResponse.rawHeaders[i], httpResponse.rawHeaders[i + 1]);
```

Comment:
Same question here, is `httpResponse.rawHeaders` needed if you already have all the entries from `httpResponse.headers`?

Reply:
In any case, it's always good to add comments for non-trivialities like these.

Reply:
In the specific case of the issue it was so that the headers did not contain the all the entries of the raw headers. This is why I forced them inside in this way. I'm not sure why that would ever happen

Reply:
Which headers were not included specifically?
It's possible that this would be caused by some other bug elsewhere.